Add description to inspections.

mwz opened the pull request:
@sksamuel @mccartney I've started adding descriptions to inspections, but I wanted to check with you first if you're happy with this approach before I do this for all of the inspections. I've opened this draft with just few inspections updated as an example.

Here are the key changes:
I've added a `description` to the `Inspection` class and made the `explanation` non-optional - I think every inspection should have it. I'm also planning to remove code snippets from all the inspections and just use the description to create each warning (maybe except for some cases where the message can be more specific, e.g. the `CollectionNamingConfusion`).

It might take a while to apply those changes across all of the inspections, so I wanted to make sure you're both happy with this approach before I invest the time into this PR.

Relates to #219.

mccartney commented:
I like it!

mccartney commented:
Is there any way to cooperate with you on this one? I find it hard to guess where you are on this one - e.g. is there some range of inspections we could take care of (by adding texts)?

mwz commented:
> Is there any way to cooperate with you on this one? I find it hard to guess where you are on this one - e.g. is there some range of inspections we could take care of (by adding texts)?

Thanks for asking, @mccartney - I think this is finished now. Here's a brief summary:

1. I've updated all of the inspections with an additional explanation and I hope I also made the descriptions more consistent across the board, but please review thoroughly as it's easy to miss something or make a mistake when you're changing 170 files 😄
1. I also initially removed printing code snippets, but I realised they were quite useful, so I added them back in, but I separated them from the descriptions.
1. I've tried to make as little changes as possible, but you'll probably notice a tiny tiny bit of reformatting (but it should be mostly just whitespaces/new lines). (I'd like to add scalafmt to this project and autoformat the entire codebase in a separate PR if that's ok with you guys.)
1. I removed all of the empty/non-implemented inspections, which I will raise separate issues for so we can keep track of should anyone want to implement them in the future.
1. I removed `2.13.0` from the build matrix in travis as it looks like the build is failing for this version (I'm assuming due to some sort of regression in 2.13.0 that was fixed in 2.13.1?). I think since the artifacts are not published for 2.13, we're safe to remove it from the build. Any thoughts? I can see you raised a separate issue about it @mccartney (#298). 
1. I've come across some false positives for `VarClosure` when testing a snapshot against some other projects - I'll raise a separate issue about it as it's not related to this PR (and this has not been released yet as far as I can see).
1. I want to do a little bit more testing to check a few other things before I can say I'm happy with these changes, but I'll mark it as ready for review as I'm not expecting any other major changes from my end.

mwz commented:
Since #298 is now addressed, I'll add 2.13 back to the build matrix.

mccartney commented on src/main/scala/com/sksamuel/scapegoat/inspections/LonelySealedTrait.scala:
```diff
@@ -5,7 +5,12 @@ import com.sksamuel.scapegoat.{Inspection, InspectionContext, Inspector, Levels}
+  text = "Lonely sealed trait",
+  defaultLevel = Levels.Error,
+  description = "Checks for sealed traits without any classes extending it.",
+  explanation = "A sealed traits that is not extended is considered dead code."
```
```suggestion
  explanation = "A sealed trait that is not extended is considered dead code."
```

mccartney commented on src/main/scala/com/sksamuel/scapegoat/inspections/VariableShadowing.scala:
```diff
@@ -5,7 +5,12 @@ import com.sksamuel.scapegoat.{Inspection, InspectionContext, Inspector, Levels}
+  text = "Variable shadowing",
+  defaultLevel = Levels.Warning,
+  description = "Checks for multiple uses of the variable name in nested scopes.",
+  explanation = "Variable shadowing is very useful, but can easily lead to nasty bugs in your code. Shadowed variables can be potentially confusing to other maintainers when the same name is adopted to have a new meaning in a nested scope."
```
Hmm, this is surprising:
> Variable shadowing is very useful

mwz replied:
Obviously this is quite subjective like a lot of other inspections. Some languages disallow variable shadowing, some allow it in some form. I personally find it useful in small and isolated scopes where otherwise I would have to make up an arbitrary name because another identifier in the scope is already using it.

mccartney commented on src/main/scala/com/sksamuel/scapegoat/inspections/collections/AvoidSizeEqualsZero.scala:
```diff
@@ -3,7 +3,12 @@ package com.sksamuel.scapegoat.inspections.collections
+  text = "Avoid Traversable.size == 0",
+  defaultLevel = Levels.Warning,
+  description = "Checks for use of Traversable.size.",
+  explanation = "Traversable.size can be slow for some data structure, prefer Traversable.isEmpty, which is O(1)."
```
```suggestion
  explanation = "Traversable.size can be slow for some data structures, prefer Traversable.isEmpty, which is O(1)."
```

mccartney replied:
```suggestion
  explanation = "Traversable.size can be slow for some data structure, prefer .isEmpty, which is O(1)."
```

mccartney commented on src/main/scala/com/sksamuel/scapegoat/inspections/collections/AvoidSizeNotEqualsZero.scala:
```diff
@@ -3,7 +3,12 @@ package com.sksamuel.scapegoat.inspections.collections
+  text = "Avoid Traversable.size != 0",
+  defaultLevel = Levels.Warning,
+  description = "Checks for use of Traversable.size.",
+  explanation = "Traversable.size can be slow for some data structures, prefer Traversable.nonEmpty, which is O(1)."
```
```suggestion
  explanation = ".size can be slow for some data structures, prefer .nonEmpty, which is O(1)."
```

mccartney commented on src/main/scala/com/sksamuel/scapegoat/inspections/collections/JavaConversionsUse.scala:
```diff
@@ -3,8 +3,12 @@ package com.sksamuel.scapegoat.inspections.collections
+  text = "Java conversions",
+  defaultLevel = Levels.Warning,
+  description = "Checks for use of Java conversions.",
+  explanation = "Use of java conversions can lead to unusual behaviour. It is recommended to use JavaConverters."
```
```suggestion
  explanation = "Use of Java conversions can lead to unusual behaviour. It is recommended to use JavaConverters."
```

mccartney commented on src/main/scala/com/sksamuel/scapegoat/inspections/collections/NegationIsEmpty.scala:
```diff
@@ -3,7 +3,12 @@ package com.sksamuel.scapegoat.inspections.collections
+  text = "!isEmpty can be replaced with nonEmpty",
+  defaultLevel = Levels.Info,
+  description = "Checks whether !isEmpty can be replaced with nonEmpty.",
+  explanation = "!Traversable.isEmpty can be replaced with Traversable.nonEmpty to make it easier to reason about."
```
```suggestion
  explanation = "!.isEmpty can be replaced with.nonEmpty to make it easier to reason about."
```

mccartney commented on src/main/scala/com/sksamuel/scapegoat/inspections/collections/NegationNonEmpty.scala:
```diff
@@ -3,7 +3,12 @@ package com.sksamuel.scapegoat.inspections.collections
+  text = "!nonEmpty can be replaced with isEmpty",
+  defaultLevel = Levels.Info,
+  description = "Checks whether !nonEmpty can be replaced with isEmpty.",
+  explanation = "!Traversable.nonEmpty can be replaced with Traversable.isEmpty to make it easier to reason about."
```
```suggestion
  explanation = "!.nonEmpty can be replaced with.isEmpty to make it easier to reason about."
```

mwz commented:
Thanks, @mccartney for fixing all the typos.
